### ISSUE TITLE
BUG?: selecting multiple tags always filters to images that MATCH both

### ISSUE THREAD

**Issue by crackfoo** [toshiaki1729/stable-diffusion-webui-dataset-tag-editor#15]
Hi, 

I'm not sure exactly how to explain or if this is by design or a limitation, but if I select a tag firsttag(10) and then secondtag(5) the end results is only the images with both of those tags, which I was thinking it would keep all images with any of those tags, not a combination of them. So when I remove those tags, they'll pop back up again with the images that were filtered out because they did have both of the tags I selected.

Basically, when I look at the list and I see tags that I KNOW aren't in my images, I'd like to be able to simply tick all the boxes of the wrong tags and remove them in the edit line... Maybe there needs to be a logic switch to choose the matching/filter strategy? Or maybe even just a switch to turn the filtering on/off?

Thanks

**Comment by toshiaki1729** [comment]
It was by design but I will add logic switch a little while later.
The previous merge changes almost nothing in behavior, but changes internal filter to prepare logic switches.

**Comment by crackfoo** [comment]
Thanks a lot! It’s just really time consuming doing one at a time and
waiting for the reloading of images when I have 3-4K of them :)
